.github/workflows/pytest_gpu.yml

```diff
@@ -0,0 +1,210 @@
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup Linux
+        if: runner.os == 'linux'
+        run: |
+          sudo apt-get update && sudo apt-get install -y libsndfile1 cmake ccache build-essential g++-8 gcc-8
+          cmake --version
+
+      - name: Install CUDA drivers
+        run: |
+          wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-ubuntu2004.pin
+          sudo mv cuda-ubuntu2004.pin /etc/apt/preferences.d/cuda-repository-pin-600
+          wget https://developer.download.nvidia.com/compute/cuda/11.5.1/local_installers/cuda-repo-ubuntu2004-11-5-local_11.5.1-495.29.05-1_amd64.deb
+          sudo dpkg -i cuda-repo-ubuntu2004-11-5-local_11.5.1-495.29.05-1_amd64.deb
+          sudo apt-key add /var/cuda-repo-ubuntu2004-11-5-local/7fa2af80.pub
+          sudo apt-get update
+          sudo apt-get -y install cuda
+        shell: bash
+
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-py${{ matrix.python-version }}-torch${{ matrix.pytorch-version }}-${{ hashFiles('requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ matrix.python-version }}-torch${{ matrix.pytorch-version }}-
+
+      - name: Install dependencies
+        env:
+          HOROVOD_WITH_PYTORCH: 1
+          HOROVOD_WITHOUT_MPI: 1
+          HOROVOD_WITHOUT_TENSORFLOW: 1
+          HOROVOD_WITHOUT_MXNET: 1
+        run: |
+          python --version
+          pip --version
+          python -m pip install -U pip
+          cmake --version
+
+          echo "MARKERS:" $MARKERS
+
+          if [ "$PYTORCH" == "nightly" ]; then
+            cat requirements.txt | sed '/^torch[>=<]/d' | sed '/^torchtext[>=<]/d' | sed '/^torchvision[>=<]/d' | sed '/^torchaudio[>=<]/d' > requirements-temp && mv requirements-temp requirements.txt
+            pip install --pre torch torchtext torchvision torchaudio
+          else
+            pip install torch==$PYTORCH torchtext torchvision torchaudio
+          fi
+          pip install protobuf==3.20.1 # https://github.com/databrickslabs/dbx/issues/257
+
+          if [ "$RAY_VERSION" == "nightly" ]; then
+            # NOTE: hardcoded for python 3.9 on Linux
+            pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp39-cp39-manylinux2014_x86_64.whl
+          else
+            # installing `six` early resolves ModuleNotFound error in ray==2.1.0
+            pip install six
+            pip install ray==$RAY_VERSION
+          fi
+          ray_expected=$(python -c "import ray; print(ray.__version__)")
+
+          torch_expected=$(python -c "import torch; print(torch.__version__)")
+
+          pip install --no-build-isolation --no-use-pep517 ConfigSpace # temporary fix: https://github.com/automl/ConfigSpace/issues/173
+          pip install '.[test]'
```

Comment:
`--extra-index-url https://download.pytorch.org/whl/cu116` too to be safe.

Reply:
Might need to run this step from the dockerfile before horovod gets installed https://github.com/ludwig-ai/ludwig/blob/c3c92e3e602bdbf151f3fde3970690f5a310ca43/docker/ludwig-ray-gpu/Dockerfile#L49

Reply:
Got it, adding that now and seeing how it affects the test. Thanks!

Reply:
Looks like everything went just fine 😄 resolving this conversation.